cloudflare/resource_cloudflare_custom_ssl_test.go

```diff
@@ -0,0 +1,104 @@
+    private_key = "-----BEGIN PRIVATE KEY-----\nMIIEvwIBADANBgkqhkiG9w0BAQEFAASCBKkwggSlAgEAAoIBAQDNFL3UKnlP9Ca/\nEHlahf8EWR/so9SgZkzlKluZT+coNIb7wfTI04/ZjJzEnVmVZbp02yBsFjVjpc28\nFg5QgOSU27npFlMZR9QwJBAeWozTjEJvWhmafByaqoTLdbEQdAcY2l8BGaAPumsM\n/hdwWzHIsEIE7pTv/BtDVII/gs6U6LoSClNh0UgFrwRTp6/RRtoCNfIXjXV8Xg/0\nqBE6ieyzpKOZLOwSujzslXzSQwBqtNoR6eskxjL4ORLnmsQW0CyEn+i/ce9UHOAm\nX22JUHEX/7HJsza6VZHvyYiL1vK/lajOt0IUpQbGtRs/Ou/gL9mD2m1kwevY/fq2\nGaCXXMCLAgMBAAECggEBAIvvvkQ6o0KiV5oCNLxHOKcP5Y/Ejr7Qb2HkEFLByfqO\nNRku1MgATGTm5MXolIszuhIov6vhT5bqOUNBTY0zFkZY1DevSw6yC6C5yuHbacKk\nL2Tp9xSJ4b7L4gcvDJ4sffdAcpk+khCJZKid7QJ2x7aoRrQ01B4ZScUcsi+CI1JJ\nbkp3LvH+pYfz2/NXGwqsvgxKAIbMTsSv271BWTm0Pr/cQ3AECnKCWnS0/3LNy1SP\nt1xXAuL8vVfkl84RqTYws3cXQBvwo9KobqZFdNvIsGbyIdV4Z8T4KG4QzS2DBFrh\nFzOKBw+uvkBMjJCeBAd1Daha7ylJio+zOvN9j5RGNFkCgYEA7x8FcO4yO6bfTGrd\n/ocbaovSsABHkxmEZb6Ki7EPlQgmrjwmUaNUf/pwoW/RC58d+gHrcBkw8Y9DnUvL\njy9sljKJS+ApS16uILelQ556A0GCYLla5vJ/oDAuJ8kAtxCkPvvdFlHP6GDuODsY\nl6pauLp2VuoOa9IeVs8xlQPiwt0CgYEA246bMyC/gQYIEQj4sIqClUd60mMhjupN\nkbQdMsOjM8sBjf9s3ZIONWXI5UFE/QrdDdMsZBP4xUvX/CgcoXjxsRoIyWFR/vHC\n++o9yKj9XfFP5g41U7eOJj36GiVZ2s/J1qglJ44cFmIBaReGoXopS/ccOTcScDbK\nPOIOArcdFocCgYEAsmhhxeVig1E476oYYaxqTy9tjbVXsa/rMYJdmmYL6zS+r2bf\nbC/Bfw7a9AgaX2JjmkHOaL/S3Zf3aafAg99tVA72ky73gG1u26hJXM8j18QLw6Dn\n6sHpaRophbOZnfyDnx6J0PpPdeDEPB4Tdi07LPKqEqTlB5so2boTE0xn5t0CgYEA\nn9WbSodGoskfSjd7xBmxorccxNiB76bGvZGfx/sAbo4VHaibOlo/mcP1kmAHtycX\nch8Pq/OWIRtrqxgQb8S6PrGzP9dnd+/MgNQwEkpj2OX5woMJc16nT1PDJRGX7mFi\nkLBsC/W6oNjMKhOEYT2rnq/Qjh53f9WDOPtgM73WoTUCgYAnYT8lyRyRVrgm925h\nrzE1uHcK7k6ZgSsIHTy3HLi7GQTzIbAMDmz0ZrTZgyEnrvTSuvHklxreUrBBt1+W\n5HbyCTwlUxkKQcj7QX5IlKe1IpvhSjvplG6bPHvHQWPmnFL4Q0WPRwISWPcoP/Ih\nqHXCFswfipntRh3ghFnACt55xg==\n-----END PRIVATE KEY-----\n"
+    bundle_method = "ubiquitous",
+	geo_restrictions = {
```

Comment:
After fixing the issue above, this is now failing with the following error

```
=== RUN   TestAccCloudflareCustomSSL_Basic
=== PAUSE TestAccCloudflareCustomSSL_Basic
=== CONT  TestAccCloudflareCustomSSL_Basic
--- FAIL: TestAccCloudflareCustomSSL_Basic (0.02s)
    testing.go:568: Step 0 error: config is invalid: Incorrect attribute value type: Inappropriate value for attribute "custom_ssl_options": element "geo_restrictions": string required.
FAIL
FAIL	github.com/terraform-providers/terraform-provider-cloudflare/cloudflare	0.050s
```

We will need to look at getting these fixed before we merge this in.

Reply:
Also, the value `usd` is invalid here. The schema only allows `"us", "eu", "highest_security"`

Reply:
Fixed as well

Reply:
I (and integration tests) are still seeing failures here. It doesn't look like you've addressed the issue with `geo_restrictions` yet.